File: client/Assets/UIDataBinding/databinding/cccomps/CCDataBindHub.ts
import { ccclass } from "../../convenient";
import { DataBindHub } from "../pure/DataBindHub";
import { CCMyComponent } from "./CCMyComponent";
import { DataBindHubHelper } from "./DataBindHubHelper";
/**
 * 数据绑定集线器
 */
@ccclass("CCDataBindHub")
export class CCDataBindHub extends CCMyComponent {
	dataBindHub: DataBindHub = new DataBindHub()

	protected needAttach() {
		return this.isAttachCalled && this.enabled && !!this.node.parent?.activeInHierarchy
	}

	integrate() {
		DataBindHubHelper.onAddDataBindHub(this)
	}

	deintegrate() {
		this.dataBindHub.clear()
	}

	/**
	 * 集成
	 * - 遍历所有浅层子hub, 设置父节点为自身
	 */
	relate() {
		DataBindHubHelper.onRelateDataBindHub(this)
	}

	derelate() {
		DataBindHubHelper.onDerelateDataBindHub(this)
	}

	protected onPreload() {
		this.integrate()
	}
	protected onPreDestroy() {
		this.deintegrate()
	}

	onAttach() {
		this.relate()
	}

	onDeattach() {
		this.derelate()
	}

}

File: client/Assets/UIDataBinding/databinding/cccomps/CCDataBindHub.cs

using UnityEngine;
using System;
using System.Collections.Generic;


namespace UI.DataBinding
{
	public class CCDataBindHub : CCMyComponent, ICCDataBindHub
	{
		public virtual DataBindHub dataBindHub { get; set; } = new DataBindHub();

		protected override bool needAttach()
		{
			return this.isAttachCalled && this.enabled && !!this.transform.parent.IsActiveInHierarchy();
		}

		public virtual void integrate()
		{
			DataBindHubHelper.onAddDataBindHub(this);
		}

		public virtual void deintegrate()
		{
			this.dataBindHub.clear();
		}

		/**
		 * 集成
		 * - 遍历所有浅层子hub, 设置父节点为自身
		 */
		public virtual void relate()
		{
			DataBindHubHelper.onRelateDataBindHub(this);
		}

		public virtual void derelate()
		{
			DataBindHubHelper.onDerelateDataBindHub(this);
		}

		protected override void onPreload()
		{
			this.integrate();
		}
		protected override void onPreDestroy()
		{
			this.deintegrate();
		}

		protected override void onAttach()
		{
			this.relate();
		}

		protected override void onDeattach()
		{
			this.derelate();
		}

	}
}
